attribution flag now resizes with font-size changes

sumitsaurabh927 opened the pull request:
Tested on Chrome on Fedora and things are working fine. I changed font-size in Chrome's console and the attribution flag changed accordingly.

Malvoz commented on src/control/Control.Attribution.js:
```diff
@@ -6,7 +6,7 @@ import * as DomEvent from '../dom/DomEvent';
 import Browser from '../core/Browser';
 
-var ukrainianFlag = '<svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="12" height="8"><path fill="#4C7BE1" d="M0 0h12v4H0z"/><path fill="#FFD500" d="M0 4h12v3H0z"/><path fill="#E0BC00" d="M0 7h12v1H0z"/></svg>';
+var ukrainianFlag = '<svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="12" height="8" + viewBox="0 0 12 8" class="leaflet-attribution-flag"><path fill="#4C7BE1" d="M0 0h12v4H0z"/><path fill="#FFD500" d="M0 4h12v3H0z"/><path fill="#E0BC00" d="M0 7h12v1H0z"/></svg>';
```
Copy paste error? `+` Should not be included here.

Malvoz reviewed (changes requested):
IE treats `height: auto` on the SVG differently from modern browsers such as Chrome and Firefox, resulting in a visual bug. Will need to resolve this.

sumitsaurabh927 commented:
I've removed `+` sign in the latest commit. Unfortunately I don't have access to IE. How else can I contribute? When will my PR be merged?

mourner reviewed (changes requested):
Please only add the necessary changes to fix the bug, don't make unrelated changes — the `vertical-align` was there for a reason but was removed, and `!important` is necessary too. Also please keep formatting consistent.

sumitsaurabh927 commented:
I'm so sorry. I've fixed my mistakes and added back `vertical-align` and `!important`. I'm presuming the comment was the issue with consistency and have removed it. Shall I push these changes?

Falke-Design commented:
Yes always push your changes if you are sure that they are ready to review

sumitsaurabh927 commented:
Done. Thanks for the tip!

mourner commented on dist/leaflet.css:
```diff
@@ -423,10 +423,12 @@ svg.leaflet-image-layer.leaflet-interactive path {
 	text-decoration: underline;
 	}
-.leaflet-control-attribution svg {
+.leaflet-attribution-flag{
```
```suggestion
.leaflet-attribution-flag {
```

mourner reviewed (commented):
The changes looks good now — let's see if we can find a way to resolve the issue in IE, maybe @Malvoz has ideas?

sumitsaurabh927 commented on dist/leaflet.css:
```diff
@@ -423,10 +423,12 @@ svg.leaflet-image-layer.leaflet-interactive path {
 	vertical-align: baseline !important;
-	}
+	width: 1em;
+	height: auto;
```
Done and pushed!

Malvoz replied:
The height value in https://github.com/Leaflet/Leaflet/pull/8183/commits/c554d64f8e5ca4414983f68f5872cbbe350ff150 computes to `8.025px` (on normal document font size, on my device). Don't know if this is bothering anyone. The suggested value here would compute to `8px` in the same scenario.

@mourner?

Malvoz replied:
@sumitsaurabh927 Did you mean to change the value from the [suggested](https://github.com/Leaflet/Leaflet/pull/8183#discussion_r857570861) one?

sumitsaurabh927 replied:
I originally set it to auto but you [suggested](https://github.com/Leaflet/Leaflet/pull/8183#discussion_r857570861) it to be `0.669em` so I set it to that. ( I thought it fixed height bug in IE)

Malvoz replied:
I suggested `0.6669em`, but you set it to `0.669em`.

sumitsaurabh927 commented:
I'm so sorry for this! Stupid mistake 🤦‍♂️
Rectified it and pushed!

sumitsaurabh927 commented on dist/leaflet.css:
```diff
@@ -423,10 +423,12 @@ svg.leaflet-image-layer.leaflet-interactive path {
-	}
+	width: 1em;
+	height: 0.6669em;
+}
```
Sorry this escaped my attention, fixed and pushed!

mourner commented:
Thank you!

sumitsaurabh927 commented:
Thank you! This was my first contribution to an open source and needless to say I'm beyond happy that it has been merged. I'm so sorry for all the silly mistakes I did and promise that I'll keep them in mind whenever I'm contributing to any OS project.
Thanks you! ❤️

mourner commented:
@sumitsaurabh927 no worries at all, you don't have to apologize so much — even contributors with decades of experience can easily make mistakes in PRs, and yours were very minor. You did well!

sumitsaurabh927 commented:
Okay. Thank you! 😊
